common/thorhelper/thorcommon.cpp

```diff
@@ -1121,18 +1205,46 @@ class CRowStreamReader : implements IExtRowStream, public CSimpleInterface
             *crcout = crccb.crc;
     }
 
-    offset_t getOffset()
+    virtual offset_t getOffset() override
+    {
+        return offset;
+    }
+
+    virtual offset_t getLastRowOffset() override
     {
-        return source.tell();
+        return currentRowOffset;
     }
 
-    virtual unsigned __int64 getStatistic(StatisticKind kind)
+    virtual unsigned __int64 getStatistic(StatisticKind kind) override
     {
         if (fileio)
             return fileio->getStatistic(kind);
         return 0;
     }
+    virtual bool setFiltersAndTranslation(unsigned numFilters, IFieldFilter **filters, IOutputMetaData *_actualFormat, IOutputMetaData *projectedFormat, RecordTranslationMode translationMode, unsigned expectedCrc, unsigned daliCrc) override
+    {
+        actualFormat.set(_actualFormat);
+        actualFilter.clear();
+        fieldFilters.kill();
```

Comment:
Why are you maintaining both fieldFilters and actualFilter?

I think this code could be simplified, possibly using the "modify in place" version of the keyedTranslator->translate() method that was previously marked as deprecated but I changed my mind and it is no longer...